pkg/asset/installconfig/aws/basedomain.go

```diff
@@ -95,5 +114,21 @@ func GetPublicZone(sess *session.Session, name string) (*route53.HostedZone, err
 	if res == nil {
 		return nil, errors.Errorf("No public route53 zone found matching name %q", name)
 	}
+
 	return res, nil
 }
+
+// GetPublicZoneByID returns a public route53 zone from the zone ID
+func GetPublicZoneByID(zoneID string) (*route53.HostedZone, error) {
+	session, err := GetSession()
```

Comment:
I don't think we should have any function calling GetSession because that doesn't have any region, aws service endpoints configuration setups.

accept a session like https://github.com/openshift/installer/pull/2764/files#diff-d8bb3678152879e136aec3bd82d820b2R98

Reply:
This function has been removed since it's not needed anymore.